romancal/refpix/data.py

```diff
@@ -0,0 +1,637 @@
+from __future__ import annotations
+
+import abc
+from itertools import islice
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from roman_datamodels.datamodels import RampModel, RefpixRefModel
+
+from dataclasses import dataclass
+from enum import IntEnum
+
+import numpy as np
+from astropy import units as u
+from scipy import fft
+
+
+class Const(IntEnum):
+    """
+    The necessary assumed values for the reference pixel correction
+        - REF: the number of columns used by the detector reference pixels
+        - PAD: the number of columns to add to introduce the appropriate delay
+               between channel reads from the guide window scan and other processes
+               this enables uniform time sampling of the data.
+        - CHAN_WIDTH: the number of columns in a channel (detector amplifier)
+        - N_CHANNELS: the number of channels for the detector
+        - N_COLUMNS: the total number of columns in the detector
+    """
+
+    REF = 4
+    PAD = 12
+    CHAN_WIDTH = 128
+    N_DETECT_CHAN = 32
+    N_COLUMNS = CHAN_WIDTH * N_DETECT_CHAN
```

Comment:
Would it be better to document the constants directly, not in the class string, i.e.
```
REF = 4
# The number of columns used by the detector reference pixels
```

Reply:
No, I don't think that would be a good idea in this case. The constants enum should be pulled into the documentation automatically (I don't think I have actually written the documentation yet for refpix), meaning the doc string will get incorporated into the docs. Using a comment will not be pulled into the documentation.

Note that the reason for creating constants this way, is so that they are immutable. If they were not wrapped into the enum, there is the potential risk of them being altered by mistake.

Reply:
The question then to @schlafly and @tddesjardins is - do you envision end users wanting to change any of the constants so that they can explore different options of the step?

Reply:
I don't think allowing this is a very safe idea. There are a lot of baked in assumptions about how the computation is structured which are related to these values. While the code always references these constants and I tried to make the code independent of these constants, there are subtle assumptions about the relationships among the constants which if violated could cause unexpected behavior of the correction. Hence, my efforts to make the constants actually constant.

Reply:
These constants are properties of the detector and will not change.  Maybe one could imagine changes to the readout mode changing PAD.  But that would be a pretty major change to the detector configuration and isn't something the user would want to change.  I don't expect we'll need to change it.